File: app/src/main/java/com/dumuzeyn/mp3player/PlaybackActions.java
package com.dumuzeyn.mp3player;

import java.util.List;

/** Commands exposed to playback UI without exposing MediaController internals. */
interface PlaybackActions {
    void playTrack(Track track);

    void playTracks(List<Track> tracks, boolean shuffle);

    void togglePlayPause();

    void next();

    void previous();

    void seekTo(long positionMs);

    void cycleRepeatMode();
}

File: app/src/main/java/com/dumuzeyn/mp3player/PlaybackActions.kt
package com.dumuzeyn.mp3player

/** Commands exposed to playback UI without exposing MediaController internals. */
interface PlaybackActions {
    fun playTrack(track: Track)
    fun playTracks(tracks: List<Track>, shuffle: Boolean)
    fun togglePlayPause()
    fun next()
    fun previous()
    fun seekTo(positionMs: Long)
    fun cycleRepeatMode()
}
